readthedocs/filetreediff/__init__.py

```diff
@@ -69,10 +69,15 @@ def get_diff(version_a: Version, version_b: Version) -> FileTreeDiff | None:
         if file_a.main_content_hash != file_b.main_content_hash:
             files_modified.append(file_path)
 
+    def sortpath(filename):
+        # Sort the files based on the amount of /.
+        # It will show the top level filenames first.
+        return filename.count("/")
```

Comment:
This is not the best sorting, but I think it works for now.

I want to sort the files by deepness/hierarchy, but also alphabetically ascending. The last part is missing in this sorting output.

If you know a simple way to do this, I can update the PR with those changes 👍🏼

Reply:
You can sort like this, but that will put all deep directories first

```suggestion
    def sortpath(filename):
        return filename.split("/")
```

or you can sort like this, that will put the less deep directories first, but it won't group the results by directory.


```suggestion
    def sortpath(filename):
        parts = filename.split("/")
        return len(parts), parts
```

You can also first sort by filenames, and then by directories

```python
files = [Path(file) for file in files]
groups = {}
for file in files:
    groups.setdefault(file.parent, []).append(file)


sorted_files = []
for dir in sorted(groups.keys()):
    groups[dir].sort()
    sorted_files.extend(groups[dir])
```

But I think the second option is nicer to look at.

Reply:
I went ahead and commited the second option.